src/main/java/org/wso2/siddhi/plugins/idea/debugger/SiddhiDebugProcess.java

```diff
@@ -0,0 +1,687 @@
+                    (myDebugFilePath)) {
+                return;
+            }
+            PsiFile psiFile = PsiManager.getInstance(getSession().getProject()).findFile(file);
+            int offset = breakpointPosition.getOffset();
+            PsiElement element = null;
+            if (psiFile != null) {
+                element = psiFile.findElementAt(offset);
+            }
+            if (element == null) {
+                return;
+            }
+            PsiElement nextVisibleSibling = getNextVisibleSiblingSkippingComments(element);
+            if (!(((LeafPsiElement) element).getElementType() == SiddhiTypes.FROM
+                    //TODO:once the antlr tree collapsing issue fixed remove one getParent() from below
+                    && element.getParent().getParent().getParent() instanceof ExecutionElementNode
+                    && PsiTreeUtil.getParentOfType(element, QueryNode.class) != null
+                    && PsiTreeUtil.getParentOfType(element, PartitionNode.class) == null
+                    && PsiTreeUtil.getParentOfType(element, AnonymousStreamNode.class) == null
+                    && PsiTreeUtil.getParentOfType(nextVisibleSibling, QueryInputNode.class) != null)) {
+                return;
+            }
+            inBreakpoints.add(breakpoint);
+            sendBreakpoints();
+            getSession().updateBreakpointPresentation(breakpoint, AllIcons.Debugger.Db_verified_breakpoint, null);
+        }
+
+        @Override
+        public void unregisterBreakpoint(@Nonnull XLineBreakpoint<SiddhiBreakpointProperties> breakpoint,
+                                         boolean temporary) {
+            XSourcePosition breakpointPosition = breakpoint.getSourcePosition();
+            if (breakpointPosition == null) {
+                return;
+            }
+            inBreakpoints.remove(breakpoint);
+            sendRemovedBreakpoint(breakpoint);
+        }
+
+        void sendRemovedBreakpoint(@Nonnull XLineBreakpoint<SiddhiBreakpointProperties> breakpoint) {
+            XSourcePosition breakpointPosition = breakpoint.getSourcePosition();
+            if (breakpointPosition == null) {
+                return;
+            }
+            VirtualFile file = breakpointPosition.getFile();
+            if (!file.getPath().replaceAll("/", Matcher.quoteReplacement(File.separator)).equalsIgnoreCase
+                    (myDebugFilePath)) {
+                return;
+            }
+            int line = breakpointPosition.getLine();
+            String name = file.getName();
+            String terminal = "IN";
+            int queryIndex = -1;
+            for (int j = 0; j < queryInLinePositions.size(); j++) {
+                if ((queryInLinePositions.get(j) - 1) == line) {
+                    queryIndex = j;
+                    break;
+                }
+            }
+            String stringBuilder = "{\"command\":\"" + Command.REMOVE_BREAKPOINT +
```

Comment:
Why the name stringBuilder? Use a `StringBuilder` instead of string concatenation or use an object and use GSON to generate the string.

Reply:
Fixed in commit 3aa87a6